src/api/ledgerManager.test.js

```diff
@@ -0,0 +1,280 @@
+      const localLedger = ledgerWithIdentities();
+
+      const manager = createLedgerManager(localLedger.eventLog());
+      const res = await manager.reloadLedger();
+
+      expect(res.error).toBe(null);
+      expect(res.remoteChanges).toEqual([]);
+      expect(res.localChanges).toEqual(localLedger.eventLog());
+    });
+
+    it("should load a remote ledger with existing events while preserving local ledger changes", async () => {
+      const remoteLedger = ledgerWithIdentities();
+      setRemoteLedger(remoteLedger);
+
+      const manager = createLedgerManager(remoteLedger.eventLog());
+      manager.ledger.activate(id1);
+      manager.ledger.activate(id2);
+
+      const res = await manager.reloadLedger();
+
+      const expectedLocalChanges = [
+        {
+          ledgerTimestamp: expect.anything(),
+          uuid: expect.anything(),
+          action: {type: "TOGGLE_ACTIVATION", identityId: id1},
+          version: "1",
+        },
+        {
+          ledgerTimestamp: expect.anything(),
+          uuid: expect.anything(),
+          action: {type: "TOGGLE_ACTIVATION", identityId: id2},
+          version: "1",
+        },
+      ];
+
+      expect(res.error).toBe(null);
+      expect(res.remoteChanges).toEqual([]);
+      expect(res.localChanges).toEqual(expectedLocalChanges);
+      expect(manager.ledger.eventLog()).toEqual([
+        ...remoteLedger.eventLog(),
+        ...expectedLocalChanges,
+      ]);
+    });
+
+    it("should replay local ledger events on top of new remote ledger events", async () => {
+      const {manager, baseEventLog} = syncEventsFixture();
+
+      manager.ledger.transferGrain({
+        from: id1,
+        to: id2,
+        amount: g("4"),
+        memo: "local transfer",
+      });
+
+      const expectedRemoteChanges = [
+        {
+          ledgerTimestamp: expect.anything(),
+          uuid: expect.anything(),
+          version: "1",
+          action: {
+            type: "TRANSFER_GRAIN",
+            amount: "6",
+            memo: "remote transfer",
+            from: id1,
+            to: id2,
+          },
+        },
+      ];
+
+      const expectedLocalChanges = [
+        {
+          ledgerTimestamp: expect.anything(),
+          uuid: expect.anything(),
+          version: "1",
+          action: {
+            type: "TRANSFER_GRAIN",
+            amount: "4",
+            memo: "local transfer",
+            from: id1,
+            to: id2,
+          },
+        },
+      ];
+
+      const res = await manager.reloadLedger();
+
+      expect(res.error).toBe(null);
+      expect(res.remoteChanges).toEqual(expectedRemoteChanges);
+
+      expect(res.localChanges).toEqual(expectedLocalChanges);
+
+      expect(manager.ledger.eventLog()).toEqual([
+        ...baseEventLog,
+        ...expectedRemoteChanges,
+        ...expectedLocalChanges,
+      ]);
+    });
+
+    it("should prevent replaying conflicting changes on remote ledger and return error", async () => {
```

Comment:
Due to other comment, would you add a test case where the first replay succeeds and the second replay fails?